pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -880,6 +881,11 @@ func PodMatchesNodeSelectorAndAffinityTerms(pod *v1.Pod, node *v1.Node) bool {
 			// if nodeAffinity.RequiredDuringSchedulingRequiredDuringExecution == nil && nodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution == nil {
 			return true
 		}
+		for _, existingPod := range nodeInfo.Pods() {
```

Comment:
It doesn't make sense to introduce an O(n) operation in a critical path for working around a corner issue in kubelet.

Reply:
The O(n) operation is not the way to go, agreed.